src/core/regimes.rkt

```diff
@@ -172,8 +173,8 @@
 (define (valid-splitindices? can-split? split-indices)
   (and
    (for/and ([pidx (map si-pidx (drop-right split-indices 1))])
```

Comment:
Might as well add an `in-list` here. Also possibly there's a clever way to avoid the `drop-right` here; that copies the whole list so is a bunch of unnecessary allocation. Also I bet instead of doing `map si-pidx` we could just call `si-pidx` as necessary, and avoid copying the list again.